client/src/components/Ticketing/ticketingmanager/Doorlistpage/doorlist.tsx

```diff
@@ -82,86 +86,189 @@ const DoorList = () => {
+        console.error(error.message);
+      }
+    };
+    fetchEvents();
   }, []);
 
+  const handleEventChange = (event) => {
+    const eventId = parseInt(event.target.value);
+    setSelectedEventId(eventId);
+
+    const matchingEvents = eventListFull.filter((e) => e.eventid === eventId);
+    setAvailableTimesEvents(matchingEvents);
+  };
+
+  const handleTimeChange = (event) => {
+    const eventInstanceID = parseInt(event.target.value);
+    getDoorList(eventInstanceID);
+  };
+
+  const CustomToolbar = () => {
+    const apiRef = useGridApiContext();
+
+    const handleExport = (options: any) => {
+      apiRef.current.exportDataAsCsv(options);
+    };
+
+    return (
+      <div className='container columns-3 gap-4 ml-3 mt-3 '>
+        <button
+          className='flex flex-row  text-blue-600 gap-2
+          items-center hover:text-blue-500 px-3 py-2 rounded-lg'
+          onClick={handleExport}
+        >
+          <svg xmlns="http://www.w3.org/2000/svg" className="h-6 w-6" fill="none" viewBox="0 0 24 24" stroke="currentColor" strokeWidth={2}>
+            <path strokeLinecap="round" strokeLinejoin="round"
+              d="M7 16a4 4 0 01-.88-7.903A5 5 0 1115.9 6L16 6a5 5 0 011 9.9M9 19l3 3m0 0l3-3m-3 3V10" />
+          </svg>
+          Export
+        </button>
+      </div>
+    );
+  };
 
   const getDoorList = async (event) => {
     try {
       const token = await getAccessTokenSilently({
         audience: process.env.REACT_APP_ROOT_URL,
         scope: 'admin',
       });
-
-      const getuser = event.target.value;
-      const response = await fetch(process.env.REACT_APP_API_1_URL + `/doorlist?eventinstanceid=${getuser}`, {
+      const response = await fetch(process.env.REACT_APP_API_1_URL + `/doorlist?eventinstanceid=${event}`, {
         headers: {
-          Authorization: `Bearer ${token}`,
+          'Content-Type': 'application/json',
+          'Authorization': `Bearer ${token}`,
         },
       });
-      const jsonData = await response.json();
-      console.log(jsonData.data);
-
-      // doorlistData.data {id: custid, name, vip, donor: donorbadge, accomodations: seatingaccom, num_tickets, checkedin, ticketno }
-      setDoorList(jsonData.data);
-      setEventName(jsonData.data[0].eventname);
-      setDate(dayMonthDate(jsonData.data[0].eventdate));
-      setTime(militaryToCivilian(jsonData.data[0].eventtime));
+      const jsonRes = await response.json();
+      const jsonData = jsonRes.data;
+      const doorListData = jsonData.map((item, index) => {
+        const row = item.row.slice(1, -1).split(',');
+        if (!row || !row[0]) { // Check if the value in column 0 is present
+          setTicketsSold(false);
+          return null; // Exit early if no tickets sold
+        }
+        setTicketsSold(true);
+        return {
+          id: index,
+          firstname: row[1],
+          lastname: row[2],
+          num_tickets: row[12],
+          // TODO: hook arrived with redeemed vaue
```

Comment:
Do you happen to know what this TODO is?

Reply:
if we want to add functionality to check people into the database we'll have to add an API call when the 'arrived' box is checked. i removed the "TO DO" comment and will make a new issue/feature request to track it